src/wp-includes/option.php

```diff
@@ -1133,6 +1137,63 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ *  Determine the appropriate autoload value based on input.
+ *  Checking the size of an option value and compare it to the maximum allowed size.
+ *
+ *  This function checks the provided autoload value and returns a standardized
+ *  value ('yes', 'no', 'default-yes', or 'default-no') based on specific conditions.
+ *
+ * Check the size of an option value and compare it to the maximum allowed size.
+ *
+ * This function serializes the given value and checks its size against the maximum allowed
+ * option size defined by the 'max_option_size' filter.
+ *
+ * @param string $option The name of the option.
+ * @param mixed  $value  The value of the option to be checked.
+ *
+ * @return string Returns 'default-no' if the size exceeds the maximum allowed size, or 'default-yes' otherwise.
+ *@since 6.5.0
+ * @access private
+ *
+ */
+function determine_option_autoload_value( $option, $value, $autoload ) {
+
+	// Check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'yes' : 'no';
+	}
+
+	switch ( $autoload ) {
+		case 'no':
+		case 'yes':
+		case 'default-yes':
+		case 'default-no':
+			return $autoload;
+
+	}
+	// Serialize the value and check its size against the maximum allowed option size.
+	$serialized_value = maybe_serialize( $value );
+	$size             = strlen( $serialized_value );
+
+	/**
+	 * Filters the maximum size of option value in bytes.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param int    $max_option_size The option-size threshold, in bytes. Default 150000.
+	 * @param string $option          The name of the option.
+	 * @param mixed  $value           The value of the option.
+	 */
+	$max_option_size = (int) apply_filters( 'max_option_size', 150000, $option, $value );
```

Comment:
3 points of feedback:
* Maybe better call this something like `wp_max_autoloaded_option_size`?
* The `@since` annotation needs an update.
* I don't think we should include the `$value` as a parameter. It doesn't make sense to define the size based on the value. If there's a good reason to pass it, we can always do that later, but let's be mindful of providing as much context as needed but as little as possible.
```suggestion
	/**
	 * Filters the maximum size of option value in bytes.
	 *
	 * @since 6.5.0
	 *
	 * @param int    $max_option_size The option-size threshold, in bytes. Default 150000.
	 * @param string $option          The name of the option.
	 */
	$max_option_size = (int) apply_filters( 'wp_max_autoloaded_option_size', 150000, $option );
```